# Replace standard BouncyCastle with FIPS-validated BouncyCastle libraries
# from /usr/share/java/bouncycastle-fips/ (installed by bouncycastle-fips package):
# - bc-fips: Core FIPS provider (replaces bcprov-jdk18on)
# - bcpkix-fips: PKI/X.509 support (replaces bcpkix-jdk18on)
# - bctls-fips: TLS/JSSE provider for FIPS-compliant TLS
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 1017381..be832e1 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -53,7 +53,6 @@ versions += [
   apacheda: "1.0.2",
   apacheds: "2.0.0-M24",
   argparse4j: "0.7.0",
-  bcpkix: "1.80",
   caffeine: "3.2.0",
   bndlib: "7.1.0",
   checkstyle: project.hasProperty('checkstyleVersion') ? checkstyleVersion : "10.20.2",
@@ -146,7 +145,9 @@ libs += [
   apachedsMavibotPartition: "org.apache.directory.server:apacheds-mavibot-partition:$versions.apacheds",
   apachedsJdbmPartition: "org.apache.directory.server:apacheds-jdbm-partition:$versions.apacheds",
   argparse4j: "net.sourceforge.argparse4j:argparse4j:$versions.argparse4j",
-  bcpkix: "org.bouncycastle:bcpkix-jdk18on:$versions.bcpkix",
+  bcfips: files("/usr/share/java/bouncycastle-fips/bc-fips.jar"),
+  bcpkix: files("/usr/share/java/bouncycastle-fips/bcpkix-fips.jar"),
+  bctls: files("/usr/share/java/bouncycastle-fips/bctls-fips.jar"),
   bndlib:"biz.aQute.bnd:biz.aQute.bndlib:$versions.bndlib",
   caffeine: "com.github.ben-manes.caffeine:caffeine:$versions.caffeine",
   classgraph: "io.github.classgraph:classgraph:$versions.classgraph",
